app/src/main/java/org/thoughtcrime/securesms/blurhash/BlurHashResourceDecoder.java

```diff
@@ -0,0 +1,39 @@
+package org.thoughtcrime.securesms.blurhash;
+
+import android.graphics.Bitmap;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.bumptech.glide.load.Options;
+import com.bumptech.glide.load.ResourceDecoder;
+import com.bumptech.glide.load.engine.Resource;
+import com.bumptech.glide.load.resource.SimpleResource;
+
+import java.io.IOException;
+
+public class BlurHashResourceDecoder implements ResourceDecoder<BlurHash, Bitmap> {
+
+  private static final int MAX_DIMEN = 20;
+
+  @Override
+  public boolean handles(@NonNull BlurHash source, @NonNull Options options) throws IOException {
+    return true;
+  }
+
+  @Override
+  public @Nullable Resource<Bitmap> decode(@NonNull BlurHash source, int width, int height, @NonNull Options options) throws IOException {
+    final int finalWidth;
+    final int finalHeight;
+
+    if (width > height) {
+      finalWidth  = Math.min(width, MAX_DIMEN);
+      finalHeight = (int) (finalWidth * height / (float) width);
+    } else {
+      finalHeight = Math.min(height, MAX_DIMEN);
+      finalWidth  = (int) (finalHeight * width / (float) height);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).